components/datetime/src/provider/timezones.rs

```diff
@@ -0,0 +1,123 @@
+            }
+
+            pub fn is_empty(&self) -> bool {
+                self.0.is_empty()
+            }
+        }
+
+        impl<$lt, Q: ?Sized> std::ops::Index<&Q> for $outer
+        where
+            Q: Ord,
+            Cow<'s, str>: std::borrow::Borrow<Q>,
+        {
+            type Output = $inner;
+            fn index(&self, key: &Q) -> &Self::Output {
+                &self.0.get(key).unwrap()
+            }
+        }
+    };
+}
+
+/// An ICU4X mapping to the CLDR timeZoneNames format strings.
+/// See CLDR-JSON timeZoneNames.json for more context.
+#[derive(PartialEq, Debug, Clone)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub struct TimeZoneFormatsV1<'s> {
+    /// The hour format for displaying GMT offsets.
+    pub hour_format: Cow<'s, str>,
+    /// The localized GMT-offset format.
+    pub gmt_format: Cow<'s, str>,
+    /// The localized GMT format with no offset.
+    pub gmt_zero_format: Cow<'s, str>,
+    /// The format string for a region.
+    pub region_format: Cow<'s, str>,
+    /// The format strings for region format variants
+    /// e.g. daylight, standard.
+    pub region_format_variants: LiteMap<Cow<'s, str>, Cow<'s, str>>,
+    /// The format string to fall back to if data is unavailable.
+    pub fallback_format: Cow<'s, str>,
+}
+
+/// An ICU4X mapping to the CLDR timeZoneNames exemplar cities.
+/// See CLDR-JSON timeZoneNames.json for more context.
+#[derive(PartialEq, Debug, Clone)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub struct ExemplarCitiesV1<'s>(pub LiteMap<Cow<'s, str>, Cow<'s, str>>);
+map_access!(ExemplarCitiesV1<'s> => Cow<'s, str>: 's);
+
+/// An ICU4X mapping to the long-form generic MetaZone names.
+/// See CLDR-JSON timeZoneNames.json for more context.
+#[derive(PartialEq, Debug, Clone)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub struct MetaZoneGenericNamesLongV1<'s>(pub LiteMap<Cow<'s, str>, Cow<'s, str>>);
+map_access!(MetaZoneGenericNamesLongV1<'s> => Cow<'s, str>: 's);
+
+/// An ICU4X mapping to the short-form generic MetaZone names.
+/// See CLDR-JSON timeZoneNames.json for more context.
+#[derive(PartialEq, Debug, Clone)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub struct MetaZoneGenericNamesShortV1<'s>(pub LiteMap<Cow<'s, str>, Cow<'s, str>>);
+map_access!(MetaZoneGenericNamesShortV1<'s> => Cow<'s, str>: 's);
+
+/// An ICU4X mapping to the long-form specific MetaZone names.
+/// Specific names include time variants such as "daylight."
+/// See CLDR-JSON timeZoneNames.json for more context.
+#[derive(PartialEq, Debug, Clone)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub struct MetaZoneSpecificNamesLongV1<'s>(pub LiteMap<Cow<'s, str>, MetaZoneSpecificNamesV1<'s>>);
```

Comment:
`suggestion (follow-up-if-not-minor)`: Do we need to have the keys be strings? It seems like according to https://unicode.org/reports/tr35/tr35-dates.html#Time_Zone_Names the keys are `generic`, `daylight` and `standard` and it would compress better and get better deserialize perf and use less memory if we stored it as an enum.

Reply:
If the reason is to support `special`, I'd suggest:
```rust
enum Foo {
    Generic,
    Daylight,
    Standard,
    Special(Cow<'s, str>)
}

Reply:
So I originally had these as an enum, but Shane and I discussed that we should support other variants. 

I think if we were to do this, we would want to be able to support multiple other options, instead of just one.

```rust
enum Foo {
    Generic,
    Daylight,
    Standard,
    Other(LiteMap<Cow<'s, str>, Cow<'s, str>>),
}
```

@sffc, thoughts?

In any case, I'd like to make this change in a separate patch, given the current time constraints.

Reply:
@nordzilla used to have these as an enum, and I asked that they be changed to an identifier.  I was actually hoping for a TinyStr4 or TinyStr8, but @nordzilla deferred that problem for later.

Reply:
Here's the previous thread: https://github.com/unicode-org/icu4x/pull/512#discussion_r584096143

My comment:

> We agreed from the deep dive that generic/standard/daylight is Western-biased and we should be more general. I would prefer if you could find a way to keep these variants generic throughout the stack, rather than baking in a concept deep in the guts of ICU4X that there are only 3 ways to render a time zone name (generic, standard, and daylight).

I similarly do not like a "middle ground" where Daylight, Standard, and Generic are enums, and other things are special.  We should just treat a variant as a variant, with no special cases.  Simpler code and less bias.

Reply:
Do we know what's the distribution between keys here? Is it 95% those three, or is it more spread?

Reply:
@zbraniecki empirically speaking, it's been 100% those three from every locale I've looked at so far. 

I think this is all in the interest of supporting future variants such as `"ramadan"` like we discussed in the first deep dive.

Reply:
95% or more from those three.  But, citing statistics to justify a change that embeds Western bias into a core library component is a slippery slope.  We should use identifiers, like TinyStr4, which have many of the same advantages of enumerations, without the disadvantages.

This is a conversation that deserves a design doc.  Issue for that discussion: #596

For now, can we stick with what @nordzilla has in this PR?  I then suggest that @nordzilla file follow-up issues for the open tasks after this PR, and applying TinyStr to all these sets (time zone identifiers, meta zone identifiers, and time zone variants) should be one of them.

Reply:
Absolutely. I think we agreed to stick with raw `String`s for this release, and to move toward using `TinyStr` as an optimization, along with the BCP-47 time-zone identifiers and trying to get compact metazone identifiers into CLDR.

Reply:
Yup. I'm convinced and you're right that TinyStr solves that optimally. Thank you both for explaining to me!